docs/proposals/secrets-proposal.md

```diff
@@ -0,0 +1,133 @@
+## Proposal for supporting secrets in OpenCompose
+
+### Kubernetes Secrets
+
+A very simple secret object in Kubernetes looks like this -
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kubesec
+type: Opaque
+data:
+  username: <base64 encoded value>
+  password: <base64 encoded value>
+```
+
+Note that, as seen above, one secret can have multiple `data` values.
+
+---
+
+#### Creating a new secret:
+
+To create a new secret, a top level `secrets` key can be used as follows:
+
+ ```yaml
+version: "0.1-dev"
+
+services:
+   name: foobar
+   containers:
+   - image: foo/bar:tag
+     ports:
+     - port: 8080:80
+     env:
+     - foo=bar
+     - travis_creds:
```

Comment:
@kadel - yep, I implemented this by unmarshalling to `interface{}` and then doing the type checking.
I don't think we should change `key=value` because that is the de facto for passing in env variables in most specs, and we will need something else for defining an env secret.